modules/core/src/main/java/org/apache/ignite/internal/processors/datastructures/AcquireUnfairProcessor.java

```diff
@@ -0,0 +1,46 @@
+package org.apache.ignite.internal.processors.datastructures;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.UUID;
+
+/** EntryProcessor for lock acquire operation for unfair mode. */
+public final class AcquireUnfairProcessor extends ReentrantProcessor<UUID> {
+    /** */
+    private static final long serialVersionUID = 8526685073215814916L;
+
+    /** */
+    private UUID nodeId;
+
+    /**
+     * Empty constructor required for {@link Externalizable}.
+     */
+    public AcquireUnfairProcessor() {
+        // No-op.
+    }
+
+    /** */
+    public AcquireUnfairProcessor(UUID nodeId) {
+        assert nodeId != null;
+
+        this.nodeId = nodeId;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected LockedModified tryLock(GridCacheLockState2Base<UUID> state) {
+        return state.lockOrAdd(nodeId);
```

Comment:
tryLock must not add node to waiting queue, it just return if failed to acquire lock

Reply:
fixed (name of method confuse you)